src/SourceLink.Common.UnitTests/GenerateSourceLinkFileTests.cs

```diff
@@ -137,5 +137,39 @@ public void Errors()
 
             Assert.Null(content);
         }
+
+        [Fact]
+        public void DoesNotRewriteContentIfFileContentIsSame()
+        {
+            var tempFile = Path.GetTempFileName();
```

Comment:
Do not use `Path.GetTempFileName`. The method is unreliable.

Instead use:
```
using var temp = new TempRoot();
var tempFile = temp.CreateFile();
```

this will automatically delete it, so no need for `finally`

Reply:
I'm looking at the docs and I don't really see why this API is unreliable: https://docs.microsoft.com/en-us/dotnet/api/system.io.path.gettempfilename?view=netframework-4.8. Mind clarifying why?

Reply:
It fails after there is more than 65535 files temp created.